core/chains/evm/logpoller/log_poller.go

```diff
@@ -90,12 +113,15 @@ type logPoller struct {
 	orm                   *ORM
 	lggr                  logger.Logger
 	pollPeriod            time.Duration // poll period set by block production rate
-	finalityDepth         int64         // finality depth is taken to mean that block (head - finality) is finalized
 	keepBlocksDepth       int64         // the number of blocks behind the head for which we keep the blocks. Must be greater than finality depth + 1.
```

Comment:
Ahh interesting keepBlocksDepth does not play nice with a dynamic finalityDepth. I think we should probably just change the semantics to mean keepFinalizedBlocksDepth instead? That we we can remove the assertion that keepBlocksDepth>finalized and it will cleanly generalize to explicit finality chains. I think all it would involve is small change to pruneOldBlocks and the param documentation (keeping the 100k default is still fine imo since finalityDepths are much lower than that.

Reply:
I think this is still relevant - simpler to have this just mean keepFinalizedBlocksDepth instead and remove any special handling